Documentation for customizing snap-in configuration page (#63)
devrev/fern-api-docs: fern/docs/pages/references/snap-in-configuration.mdx

# Customizing snap-in configuration pages

The DevRev snap-ins platform allows developers to define custom configuration pages for their snap-ins. 

While the default configuration page automatically renders input fields for keyrings and inputs, there may be cases where a custom configuration page is more suitable:

- Improved user experience: Developers can design the configuration page to provide a more intuitive and streamlined experience for users setting up the snap-in.
- Advanced input handling: Custom configuration pages enable developers to handle complex input scenarios, such as fetching data from external systems to populate dropdown options or validating user input.
- Branding and styling: Developers can align the configuration page with their Snap-in's branding and style guidelines, ensuring a consistent look and feel.

> ## Defining Custom Configuration Pages

Reviewer: Sentence caps on titles. https://www.notion.so/devrev/Which-words-in-a-heading-or-title-should-be-capitalized-81984d54dd414af5b343bce1327c05bd?pvs=4
```suggestion
## Defining custom configuration pages
```

                      "value": "ticket"
                    },
                    "options": [
                      {
                        "text": {
                          "text": "Ticket",
                          "type": "plain_text"
                        },
                        "value": "ticket"
                      },
                      {
                        "text": {
                          "text": "Conversation",
                          "type": "plain_text"
                        },
                        "value": "conversation"
                      }
                    ],
                    "type": "static_select"
                  },
                  "type": "input_layout"
                }
              ],
              "submit_action": {
                "action_id": "next",
                "style": "primary",
                "text": {
                  "text": "Next",
                  "type": "plain_text"
                },
                "type": "button",
                "value": "next"
              },
              "type": "form"
            }
          ],
          "type": "card"
        }
      ]
    }
  }
}
```

> In this example, the snap-kit renders a dropdown select for choosing between "Ticket" and "Conversation," along with a "Next" button. When the "Next" button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.

Reviewer: Use code phrases to refer to items in the code sample. Use bold for UI elements like buttons.
```suggestion
In this example, the snap-kit renders a dropdown select for choosing between `Ticket` and `Conversation`, along with a `Next` button. When the **Next** button is clicked, the `user_snap_in_configuration_handler` function is invoked to process the user's selection.
```

In the snap-kit action handler, the `event.payload.action.id` can be used to determine the form being submitted and call the `snap-ins.update` API to update the configuration.

> ## Update snap-in inputs (BETA)

Reviewer: ```suggestion
## Update snap-in inputs (beta)
```

Updates the inputs of a snap-in based on the inputs defined in the snap-in configuration.

> **Note: This endpoint is currently in BETA and may be subject to change in the future. Reach out to us via PLuG to subscribe to changes to BETA endpoints.**

Reviewer: ```suggestion
**Note: This endpoint is currently in beta and may be subject to change in the future. Reach out to us via PLuG to subscribe to changes to beta endpoints.**
```

> ### Request Payload

Reviewer: ```suggestion
### Request payload
```

  try {
    
    await devrevSDK.snapInsUpdate(payload);
    return { data: {}, message: 'Snap-in updated successfully', success: true };
  } catch (error: any) {
    if (error.response) {
      const err = `Failed to update the snap-in. Err: ${JSON.stringify(error.response.data)}, Status: ${error.response.status}`;
      return { ...defaultResponse, message: err };
    } else {
      return { ...defaultResponse, message: error.message };
    }
  }
}
```

In this example, the `updateSnapInInputs` function takes the `snapInId` and `inputsValues` as parameters. The `inputsValues` object should contain the input names and their corresponding values as defined in the snap-in configuration.

The function constructs the payload object with the `snapInId` and `inputsValues`, and then calls the `updateSnapInInputs` function to make the POST request to the `snap-ins.update` endpoint.

If the request is successful, it returns a success response with a status code of 200 and a message indicating that the snap-in was updated successfully.

If an error occurs, it catches the error and returns an error response with an appropriate status code and an error message containing the error details.

> Please keep in mind that this endpoint is currently in BETA, and its functionality or parameters may change in future updates.

Reviewer: ```suggestion
<Note>
Note: This endpoint is currently in beta, and its functionality or parameters may change in future updates.
</Note>
```
